### ISSUE TITLE
Android.App.Dialog.DismissEvent Crashes when event is unsubscribed.

### ISSUE THREAD

**Issue by pschmitz-survey** [xamarin/xamarin-android#1342]
### Steps to Reproduce

1. Create an dialog (we experienced this issue using a AlertDialog)
2. Subscribe to the DismissEvent
3. Attempt to cleanup by unsubscribing from DissmissEvent
4. Exception occurs

### Expected Behavior
Event is successfully unsubscribed

### Actual Behavior
Exception is thrown

### Version Information
8.0

### Log File

```
System.NotSupportedExceptionUnable to activate instance of type Android.Content.IDialogInterfaceOnDismissListenerImplementor from native handle 0x7ff8e4e4c4 (key_handle 0xd1f68c3).
Raw
Java.Interop.TypeManager.CreateInstance(IntPtr handle, JniHandleOwnership transfer, Type targetType)<05e24608f1b946c7a90815988f209d76>:0
Java.Lang.Object.GetObject(IntPtr handle, JniHandleOwnership transfer, Type type)<05e24608f1b946c7a90815988f209d76>:0
Java.Lang.Object._GetObject<T>(IntPtr handle, JniHandleOwnership transfer)<05e24608f1b946c7a90815988f209d76>:0
Java.Lang.Object.GetObject<T>(IntPtr handle, JniHandleOwnership transfer)<05e24608f1b946c7a90815988f209d76>:0
Java.Lang.Object.GetObject<T>(IntPtr jnienv, IntPtr handle, JniHandleOwnership transfer)<05e24608f1b946c7a90815988f209d76>:0
Android.Content.IDialogInterfaceOnDismissListenerInvoker.n_OnDismiss_Landroid_content_DialogInterface_(IntPtr jnienv, IntPtr native__this, IntPtr native_dialog)<05e24608f1b946c7a90815988f209d76>:0
at (wrapper dynamic-method) System.Object:4be5e92e-b575-4fea-b822-20ea640e4b96 (intptr,intptr,intptr)
System.MissingMethodExceptionNo constructor found for Android.Content.IDialogInterfaceOnDismissListenerImplementor::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership)
Java.Interop.TypeManager.CreateProxy(Type type, IntPtr handle, JniHandleOwnership transfer)<05e24608f1b946c7a90815988f209d76>:0
Java.Interop.TypeManager.CreateInstance(IntPtr handle, JniHandleOwnership transfer, Type targetType)<05e24608f1b946c7a90815988f209d76>:0
```

**Comment by jonpryor** [comment]
Please attach a repro.

My *guess* is that "somewhere" you're calling `alertDialog.Dispose()`, which prevents the event unsubscription from working.

**Comment by pschmitz-survey** [comment]
Thanks for the fast response - Our implementation doesn't include an explicit dispose call.

[DialogDismiss.zip](https://github.com/xamarin/xamarin-android/files/1760511/DialogDismiss.zip)

Attaching Reproduction.

Steps:

- Tap 'Present Alert'

- Dismiss the alert by tapping outside of the alert.

- Exception/crash above occurs

**Comment by jonpryor** [comment]
I believe that this is a duplicate of https://bugzilla.xamarin.com/show_bug.cgi?id=59804 and fixed in 4269528ff90afd891ed1da97a7007842deb4be4f, present in d15-6/Xamarin.Android 8.2. (Xamarin.Android 8.1 is the current stable release; 8.2 is currently in preview/beta.)

I do not experience the crash when using current xamarin-android/master.
